pkg/collector/corechecks/snmp/session.go

```diff
@@ -86,19 +85,13 @@ func (s *snmpSession) Configure(config snmpConfig) error {
 	} else {
 		if lvl == seelog.TraceLvl {
 			traceLevelLogWriter := traceLevelLogWriter{}
-			s.gosnmpInst.Logger = stdlog.New(&traceLevelLogWriter, "", stdlog.Lshortfile)
-			s.loggerEnabled = true
+			s.gosnmpInst.Logger = gosnmp.NewLogger(stdlog.New(&traceLevelLogWriter, "", stdlog.Lshortfile))
 		}
 	}
 	return nil
 }
 
 func (s *snmpSession) Connect() error {
-	if s.loggerEnabled == false {
-		// Setting Logger everytime GoSNMP.Connect is called is need to avoid gosnmp
-		// logging to be enabled. Related upstream issue https://github.com/gosnmp/gosnmp/issues/313
-		s.gosnmpInst.Logger = nil
-	}
 	return s.gosnmpInst.Connect()
```

Comment:
Note for reviewers: the above workaround is not needed anymore since `gosnmpInst.Connect` won't try to create a new logger instance anymore:

https://github.com/gosnmp/gosnmp/pull/329/files#diff-eb6ee529a7c58196f155fd8503ee7bf31a0cec1aeb45e2262ee6319a7129bcb4R330-R331